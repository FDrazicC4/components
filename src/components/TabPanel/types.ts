import { StyledParentComponent } from '../types';

export interface Props extends StyledParentComponent {
    /**
     * The value of the matching tab component.
     */
    value: any,
}