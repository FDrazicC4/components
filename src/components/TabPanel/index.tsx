import React from 'react';
import clsx from 'clsx';
import { useTabs } from '../Tabs/context';
import { Props } from './types';

const TabPanel = React.forwardRef(({ className, value, style = {}, ...otherProps }: Props, ref: React.Ref<any>): React.ReactElement => {
    const classes = clsx(
        'brew-TabPanel',
        className,
    );

    const { value: tabValue = 0, shouldPreserveContent = false } = useTabs() || {};

    if (!shouldPreserveContent && value !== tabValue) {
        return <div />;
    }

    return (
        <div {...otherProps} 
            style={{
                ...style,
                display: !!shouldPreserveContent && value !== tabValue ? 'none' : undefined,
            }}
            className={classes} 
            ref={ref} 
        />
    );
});

export default TabPanel;