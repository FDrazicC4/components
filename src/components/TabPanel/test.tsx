import React from 'react';
import renderer from 'react-test-renderer';
import TabPanel from './index';


describe('TabPanel', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<TabPanel value={0} />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Renders correctly with property variants', () => {
        const tree = renderer
            .create((
                <React.Fragment>
                    <TabPanel value={0}>
                        children 1
                    </TabPanel>
                    <TabPanel value={0}>
                        children 2
                    </TabPanel>
                    <TabPanel value={1} />
                </React.Fragment>
            ))
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
