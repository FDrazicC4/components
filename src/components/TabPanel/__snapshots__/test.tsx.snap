// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabPanel Renders correctly with defaults 1`] = `
<div
  className="brew-TabPanel"
  style={
    Object {
      "display": undefined,
    }
  }
/>
`;

exports[`TabPanel Renders correctly with property variants 1`] = `
Array [
  <div
    className="brew-TabPanel"
    style={
      Object {
        "display": undefined,
      }
    }
  >
    children 1
  </div>,
  <div
    className="brew-TabPanel"
    style={
      Object {
        "display": undefined,
      }
    }
  >
    children 2
  </div>,
  <div />,
]
`;
